java-squid/src/main/java/org/sonar/java/model/statement/ForStatementTreeImpl.java

```diff
@@ -33,21 +31,34 @@
 import org.sonar.plugins.java.api.tree.TreeVisitor;
 
 import javax.annotation.Nullable;
+
 import java.util.Iterator;
 import java.util.List;
 
 public class ForStatementTreeImpl extends JavaTree implements ForStatementTree {
+  private final InternalSyntaxToken forKeyword;
+  private final InternalSyntaxToken openParenToken;
   private final List<StatementTree> initializer;
+  private final InternalSyntaxToken firstSemicolonToken;
   @Nullable
   private final ExpressionTree condition;
+  private final InternalSyntaxToken secondSemicolonToken;
   private final List<StatementTree> update;
+  private final InternalSyntaxToken closeParenToken;
   private final StatementTree statement;
 
-  public ForStatementTreeImpl(List<StatementTree> initializer, @Nullable ExpressionTree condition, List<StatementTree> update, StatementTree statement) {
+  public ForStatementTreeImpl(InternalSyntaxToken forKeyword, InternalSyntaxToken openParenToken, List<StatementTree> initializer,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)